src/core/unit.c

```diff
@@ -3193,6 +3193,46 @@ static int signal_name_owner_changed(sd_bus_message *message, void *userdata, sd
         return 0;
 }
 
+static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        const char *new_owner;
+        Unit *u = userdata;
+        int r;
+
+        assert(message);
+        assert(u);
+
+        r = sd_bus_message_read(message, "s", &new_owner);
```

Comment:
before reading this, you should check if the error parameter is set, which would indicate that the name is not taken right now